### ISSUE TITLE
`fatxpool`: add retracted event

### ISSUE THREAD

**Issue by michalkucharczyk** [paritytech/polkadot-sdk#5479]
Retracted event shall be sent if the best chain is switched.

_Originally posted by @bkchr in https://github.com/paritytech/project-mythical/issues/174#issuecomment-2242453807_

**Comment by skunert** [comment]
From offline discussion:
The linked comment thread references changes to only have an inclusion event, removing the need to have retracted event. RPC client users could detect themselves if their tx is included in the best chain.

**Comment by dharjeezy** [comment]
Hello @michalkucharczyk at what point is the best chain switched?

**Comment by michalkucharczyk** [comment]
In general best block is selected during block import. See this [trait](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/primitives/consensus/common/src/select_chain.rs).

From txpool perspective chain swtich happens when `tree_route` in ChainEvent has non-empty retracted path.

**Comment by dharjeezy** [comment]
Does this mean, we will have to check the tree_route from the  `NewBestBlock` event to do this, and will this be done in the maintain process for the `MaintainedTransactionPool`, [here](https://github.com/dharjeezy/polkadot-sdk/blob/master/substrate/client/transaction-pool/src/fork_aware_txpool/fork_aware_txpool.rs#L1592) ?

**Comment by michalkucharczyk** [comment]
No. This event is already handled on the View level. If you would need to enable it you need to follow this path:
https://github.com/paritytech/polkadot-sdk/blob/1bc6ca606438a65c927f14be3f36634ca0e58e8f/substrate/client/transaction-pool/src/fork_aware_txpool/multi_view_listener.rs#L374
https://github.com/paritytech/polkadot-sdk/blob/1bc6ca606438a65c927f14be3f36634ca0e58e8f/substrate/client/transaction-pool/src/fork_aware_txpool/view.rs#L155

and the event is already originated here:
https://github.com/paritytech/polkadot-sdk/blob/1bc6ca606438a65c927f14be3f36634ca0e58e8f/substrate/client/
transaction-pool/src/graph/listener.rs#L212


What is the context of your questions? Do you want to implement this? We currently decided to no implement as per https://github.com/paritytech/polkadot-sdk/issues/5479#issuecomment-2579818940

**Comment by dharjeezy** [comment]
> What is the context of your questions? Do you want to implement this? We currently decided to no implement as per [#5479 (comment)](https://github.com/paritytech/polkadot-sdk/issues/5479#issuecomment-2579818940)

Hello @michalkucharczyk thank you for your response. Yes. i want to implement this as per the issue. Do you mean there is no need to implement this again?

**Comment by michalkucharczyk** [comment]
I mean you can open the PR. But there maybe a discussion again, if we really require this.
